csrc/scheduler/normalization_utils.cpp

```diff
@@ -1613,13 +1609,36 @@ void schedulePersistentKernel(
   const auto& unroll_vectorizable_cached_tvs =
       reduction_scheduler_utils::getCachedTvsToUnrollOrVectorize(
           reference_tv, is_vectorize, cached_inputs, cached_outputs);
+
+  // For inner persistent with vectorized load, use explicity unroll for cached
+  // input if it is a persistent buffer. This won't increase register usage
+  // and encourages compiler issuing memory load instructions together. It
+  // improves performance with cuda-13.0.
+  bool use_explicit_unroll =
+      rparams->vectorize_inner_reduction && rparams->fastest_dim;
   reduction_scheduler_utils::propagateParallelization(
       reduction_tv,
       reference_tv,
       is_unroll_or_vectorization,
       use_grouped_reduction,
       reduction_tvs,
-      unroll_vectorizable_cached_tvs);
+      unroll_vectorizable_cached_tvs,
+      /*selected_tvs=*/{},
+      /*skip_input_output_unroll=*/use_explicit_unroll);
+  if (use_explicit_unroll) {
+    std::vector<TensorView*> persistent_cached_inputs;
+    for (auto tv : cached_inputs) {
+      if (std::find(persistent_buffers.begin(), persistent_buffers.end(), tv) !=
+          persistent_buffers.end()) {
+        persistent_cached_inputs.push_back(tv);
+      }
+    }
+    scheduler_utils::parallelizeAllLike(
+        reference_tv,
+        -1,
+        {persistent_cached_inputs.begin(), persistent_cached_inputs.end()},
+        {ParallelType::Unroll});
+  }
```

Comment:
Only unroll cached inputs that are persistent buffers to avoid increase register usage.